src/actions/actionCreator.js

```diff
@@ -33,11 +34,48 @@ export const loadMoviesRequest = () => ({
   type: ACTION_TYPES.LOAD_MOVIES
 });
 
+export const selectMovie = (movie) => ({
+  type: ACTION_TYPES.SELECT_MOVIE,
+  movie
+});
+
+export const clearStore = () => ({
+  type: ACTION_TYPES.CLEAR_STORE,
+});
+
+export const persistLastSearchPhrase = (lastSearchPhrase) => {
+  return ({
+    type: ACTION_TYPES.PERSIST_LAST_SEARCH_PHRASE,
+    lastSearchPhrase
+  });
+};
+
+export const LoadMovieDetailsSuccess = (movie) => ({
+  type: ACTION_TYPES.LOAD_MOVIE_DETAILS_SUCCESS,
+  movie
+});
+
+export const LoadMovieSimilarGenre = (movies) => ({
+  type: ACTION_TYPES.LOAD_MOVIES_SIMILAR_GENRE,
+  movies
+});
+
 export const buildUrl = (getState) => {
   const state = getState();
-  const url = "http://react-cdp-api.herokuapp.com/movies";
+  const url = baseURL;
   const searchBy = `&searchBy=${state.search.searchby === "title" ? "title" : "genres"}`;
-  const phrase = `?search=${state.search.phrase}`;
+  const phrase = `?search=${state.search.lastSearchPhrase}`;
+  const order = "&sortOrder=desc";
+  const limit = "&limit=15";
+
+  return `${url}${phrase}${searchBy}${order}${limit}`;
+}
+
+export const buildUrlByGenre = (getState, genre) => {
```

Comment:
Common part of URL building should be extracted to separate function

Reply:
URL building method refactored.